cloudevents/extensions/deprecation.md

```diff
@@ -0,0 +1,86 @@
+# Deprecation extension
+
+This specification defines attributes that can be included in CloudEvents to
+indicate to [consumers](../spec.md#consumer) or
+[intermediaries](../spec.md#intermediary) the deprecation of the events. These
+attributes inform consumers of CloudEvents about upcoming changes or removals,
+facilitating smoother transitions and proactive adjustments.
+
+## Notational Conventions
+
+As with the main [CloudEvents specification](../spec.md), the key words "MUST",
+"MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+"RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+However, the scope of these key words is limited to when this extension is
+used. For example, an attribute being marked as "REQUIRED" does not mean it
+needs to be in all CloudEvents, rather it needs to be included only when this
+extension is being used.
+
+## Attributes
+
+### deprecated
+
+- Type: `Boolean`
+- Description: Indicates whether the event type is deprecated. Defaults to
+  `false` if not specified.
+- Constraints
+  - OPTIONAL
+- Example: `"deprecated": "true"`
+
+### deprecationfrom
+
+- Type: `Timestamp`
+- Description: Specifies the date and time when the event type was
+  officially marked as deprecated.
+- Constraints
+  - OPTIONAL
+  - The `deprecationfrom` timestamp SHOULD remain stable once set and SHOULD
```

Comment:
Just curious, aside from this being a possible "FYI" bit of info, do people use this timestamp for some purpose in real life? Doesn't seem like it would influence people's action.... the "sunset" one is the important one.

Reply:
the `deprecationsunset` dictates the hard deadline, but `deprecationfrom` can prompt consumers to start planning their migration earlier.
IMO, it adds valuable context and encourages proactive behavior from consumers.

Reply:
In theory, but I would think the presence of any of these would indicate that the event is deprecated regardless of when the deprecation period stated, right?

Reply:
that's correct.
`deprecationfrom` adds an extra layer of clarity. This can be useful for teams who need to plan their migrations earlier rather than waiting until the sunset approaches.